index.js

```diff
@@ -0,0 +1,126 @@
+const core = require("@actions/core");
+const yaml = require("js-yaml");
+const path = require("path");
+const fs = require("fs");
+
+const jsonSchema = {
+  $schema: "http://json-schema.org/draft-07/schema#",
+  type: "object",
+  properties: {},
+  required: [],
+};
+
+function readInput() {
+  try {
+    sourceRootDir = core.getInput("source-root-dir-path");
+    return sourceRootDir;
+  } catch (error) {
+    throw new Error(`Failed to read input: ${error.message}`);
+  }
+}
+
+function readComponentYaml(filePath) {
+  try {
+    fullPath = path.join(filePath, ".choreo", "component.yaml");
+    let fileContent = fs.readFileSync(fullPath, "utf8");
+    return fileContent;
+  } catch (error) {
+    throw new Error(`Failed to read component.yaml: ${error.message}`);
+  }
+}
+
+function generateSchemaForBaseType(schema, requiredItems, type) {
+  // if required is not set or set to true, add the item to required list
+  if (schema.required === undefined || schema.required) {
+    requiredItems.push(schema.name);
+  }
+  const generatedSchema = {
+    type: type,
+  };
+  if (schema.values) {
+    generatedSchema.enum = schema.values;
+  }
+  if (schema.displayName) {
+    generatedSchema.title = schema.displayName;
+  }
+  return generatedSchema;
+}
+
+function generateSchemaFromYaml(schema, requiredItems) {
+  if (schema.type === "string") {
+    return generateSchemaForBaseType(schema, requiredItems, "string");
+  }
+
+  if (schema.type === "integer") {
+    return generateSchemaForBaseType(schema, requiredItems, "integer");
+  }
+
+  if (schema.type === "array") {
+    const generatedSchema = {
+      type: "array",
+      items: {
+        type: "string",
+      },
+      title: schema.displayName,
+    };
+    if (schema.items.type === "string") {
+      return generatedSchema;
+    }
+    if (schema.items.type === "integer") {
+      generatedSchema.items.type = "integer";
+      return generatedSchema;
+    }
+    return {
+      type: "array",
+      items: generateSchemaFromYaml(schema.items, requiredItems),
```

Comment:
What other types are we capturing here?

Reply:
items can be objects, string, integer

Reply:
then why do we need the recursion here?

Reply:
can have nested objects as well. leafs are above mentioned types